backend/prompt_studio/prompt_studio/views.py

```diff
@@ -41,3 +47,94 @@ def get_queryset(self) -> Optional[QuerySet]:
         else:
             queryset = ToolStudioPrompt.objects.all()
         return queryset
+
+    @action(detail=True, methods=["post"])
+    def reorder_prompts(self, request: HttpRequest) -> Response:
+        """Reorder the sequence of prompts based on the start and end sequence
+        numbers.
+
+        This action handles the reordering of prompts by updating their sequence
+        numbers. It increments or decrements the sequence numbers of the relevant
+        prompts to reflect the new order. If the start and end sequence numbers
+        are equal, it returns a bad request response.
+
+        Args:
+            request (HttpRequest): The HTTP request object containing the data to
+            reorder prompts.
+
+        Returns:
+            Response: A Response object with the status of the reordering operation.
+        """
+        try:
+            # Validate request data
+            serializer = ReorderPromptsSerializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+
+            # Extract validated data from the serializer
+            start_sequence_number = serializer.validated_data.get(
+                ToolStudioPromptKeys.START_SEQUENCE_NUMBER
+            )
+            end_sequence_number = serializer.validated_data.get(
+                ToolStudioPromptKeys.END_SEQUENCE_NUMBER
+            )
+            prompt_id = serializer.validated_data.get(ToolStudioPromptKeys.PROMPT_ID)
+
+            prompt_instance: ToolStudioPrompt = ToolStudioPrompt.objects.get(
+                pk=prompt_id
+            )
+            filtered_prompts_data = []
+            tool_id = prompt_instance.tool_id
+
+            # Determine the direction of sequence adjustment based on start and
+            # end sequence numbers
+            if start_sequence_number < end_sequence_number:
+                logger.info(
+                    "Start sequence number is less than end sequence number. "
+                    "Decrementing sequence numbers."
+                )
+                filters = {
+                    "sequence_number__gt": start_sequence_number,
+                    "sequence_number__lte": end_sequence_number,
+                    "tool_id": tool_id,
+                }
+                # Call helper method to update sequence numbers and
+                # get filtered prompt data
+                filtered_prompts_data = PromptStudioHelper.update_sequence_numbers(
```

Comment:
@tahierhussain  [SUGGESTION] `increment` and `filters` can be set inside the condition and `update_sequence_numbers` can be called only once after the if-else block, instead on calling inside both if and else.

Reply:
@harini-venkataraman Thanks for the suggestion. Done.